### ISSUE TITLE
UseNLog doesn't accept configuration file name

### ISSUE THREAD

**Issue by MihaMarkic** [NLog/NLog.Web#870]
**Type** (choose one): Feature request

**NLog version**: 5.0.4

**NLog.Web/NLog.Web.AspNetCore version**: 5.1.4

**NLog.Extensions.Logging version**: -

**Platform**: .Net 6

Is there a reason that UseNLog can't set config file name? I think it's hardcoded in AspNetExtensions.cs file.
I know I can use (right?) AddNLog instead, but still wonder why the it can't be set through UseNLog.
Or am I missing something?

**Comment by bakgerman** [comment]
public static ILoggingBuilder AddNLogWeb(this ILoggingBuilder builder, **string configFileName**) is in AspNetExtensions.cs file.

This method is available, but the method name is AddNLogWeb.  AddNLog will be deprecated soon.

**Comment by snakefoot** [comment]
`UseNLog()` is simple extension-method for enabling NLog Logging-Provider in a standard way that satifisfies most.

Not sure adding 10 different `UseNLog()`-methods will give the feeling of a simple standard way. Which is the case with `AddNLogWeb()`.

If you want to go your own special way, and divert from default configuration, then there is nothing stopping you.

**Comment by MihaMarkic** [comment]
@bakgerman OK, I'll go with AddNLogWeb though AddNLog works as well.
@snakefoot Image different environments - debugging, staging, production. Also, I think UseNLog already accept options argument, it's just that options don't allow file name override.

**Comment by snakefoot** [comment]
Have created a wiki-page with the available options that I could think of: https://github.com/NLog/NLog/wiki/Environment-specific-NLog-Logging-Configuration

**Comment by snakefoot** [comment]
@MihaMarkic I guess we could change `UseNLog()` so it will like `LoadConfigurationFromAppSettings()` automatically load `nlog.{environment}.config`. Would that be an improvement for you?

**Comment by MihaMarkic** [comment]
@snakefoot I've been thinking a bit more what is really required. Perhaps changes won't be needed after all. 
For other environments than development, a proper nlog.config can be copied (from whatever source file, i..e nlog.{environment}.config) into target server. So, deployment of the proper nlog.config file is up to deployment process.
For development environment one can use local only nlog.config file (excluded from source control if the file is different for different developers).

OTOH when it comes to production/staging environment - perhaps it'd make sense to use nlog.{environment}.config if server is switching environments (i.e. sometimes the same code is run as production, sometimes as staging).

TL;DR Perhaps the modification you proposed is a good step, but it should still fall back to nlog.config if specific environment file is not found.

Makes sense?

**Comment by snakefoot** [comment]
> Make sense?

Yes and you are right that the wanted fallback to `nlog.config` actually makes it a bad idea for `UseNLog()` to support loading `nlog.Production.config` since `UseNLog()` can come very late in the game, so the automatic loading of NLog-configuration have already read the default `nlog.config`.

So yes the simplest solution is to ensure the deployment-logic places the correct `nlog.config` file, that matches the install environment.

**Comment by snakefoot** [comment]
Closing question as answered due to inactivity.
